mysql/resource_ti_config_variable.go

```diff
@@ -0,0 +1,171 @@
+package mysql
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/creasty/defaults"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/tidwall/gjson"
+)
+
+func resourceTiConfigVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateOrUpdateConfigVariable,
+		Read:   ReadConfigVariable,
+		Update: CreateOrUpdateConfigVariable,
+		Delete: DeleteConfigVariable,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"pd", "tikv"}, true),
```

Comment:
I believe changing `type` will most likely need `ForceNew` at least logically. But I don't insist on this.

Reply:
yes, your are correct! I'll fix it :)

Reply:
Resolved